### PR TITLE
Add missing to device transfer

### LINKED ISSUE #1073: Device mismatch in PGD PyTorch
**Describe the bug**
When running on the GPU, PGD (PyTorch) complains about operands being on different devices:

'# Apply mask
if mask is not None:
      grad = torch.where(mask == 0.0, torch.tensor(0.0), grad)

Looks like torch.tensor(0.0) should be mapped to self.estimator.device (since mask and grad are).

https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/1.6.1/art/attacks/evasion/projected_gradient_descent/projected_gradient_descent_pytorch.py

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

This pull request adds a missing transfer to the device.

Fixes #1073 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by minhitbk — APPROVED** [review]
OK.
